dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/CycleBreakerGraphTransformer.java

```diff
@@ -52,6 +55,11 @@ private static void removeCycle(
       return;
     }
 
+    if (visited.put(node, true) != null) {
```

Comment:
you only call put and never get, so this is probably not the simplest data structure we could use here. Maybe a Set?

Reply:
I wish I have Set for object identity. JRE does not provide IdentityHashSet.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#newSetFromMap-java.util.Map- may work. Let me try this.